### DISCUSSION TITLE
Vertical align not working with tickspicture

### DISCUSSION THREAD

**Discussion by AwesomeQuest** [lvjr/tabularray#481] [Q&A]
I need to make a table of equations and figures and the equations should vertically align with the figures.
In the table below it seems that the `colspec` doesn't vertically align the $K$ with the figure, `rowspec` doesn't work either.

```LaTeX
\documentclass{article}
\usepackage{tikz}
\usepackage{tabularray}

\begin{document}

\begin{tblr}{width=0.8\textwidth,colspec={Q[c,m]|Q[c,m]}} 
    Factor & Feature \\
    $K$ & \begin{tikzpicture}
        \draw[->] (0,0) -- ++(4,0);
        \draw[red, thick] (0,1) -- ++(4,0);
        \draw (2,1) node[above]{$20\log_{10}{K}$};
        \draw (4,0) node[below]{$\omega$};
    \end{tikzpicture}\\
\end{tblr}

\end{document}
```
![image](https://github.com/lvjr/tabularray/assets/15609837/3ce6d615-c836-4bba-a8e3-9c9a1f84a542)

Thanks in advance for any pointers :)

**Comment by muzimuzhi** [comment]
Is this what you want?

```tex
\documentclass{article}
\usepackage{tikz}
\usepackage{tabularray}

\begin{document}

\begin{tblr}{width=0.8\textwidth,colspec={Q[c,m]|Q[c,m]}} 
    Factor & Feature \\
    $K$ & \begin{tikzpicture}[baseline=(k.base)]
        \draw[->] (0,0) -- ++(4,0);
        \draw[red, thick] (0,1) -- ++(4,0);
        \draw (2,1) node[above] (k) {$20\log_{10}{K}$};
        \draw (4,0) node[below]{$\omega$};
    \end{tikzpicture}\\
\end{tblr}

\end{document}
```
![image](https://github.com/lvjr/tabularray/assets/6376638/2b6db2ad-30ac-43fc-9ae8-5e3149dc1f6b)

Option `baseline=(k.base)` is added to `tikzpicture` env, in which the `k` denotes the node with text `$20\log_{10}{K}$` (see `node[above] (k) {$20\log_{10}{K}$}`).

Tikz doc for `baseline`: https://tikz.dev/tikz-scopes#pgf./tikz/baseline (unofficial online version).

**Reply by AwesomeQuest** [reply]
Thanks so much! This is exactly what I needed.
